src/utils/push.js

```diff
@@ -1,72 +1,126 @@
+import {Platform} from 'react-native';
+import {requestNotifications} from 'react-native-permissions';
 import {navigationRef} from '../../App';
-import {FEED_CHANNEL_URL} from '../constants';
 import {refreshCollection, updateHasNewNotifications} from '../redux/slices/sendbird';
-import {store} from '../redux/store';
-import {PermissionsAndroid, Platform} from 'react-native';
-import messaging from '@react-native-firebase/messaging';
+import {dispatch, store} from '../redux/store';
 
-export const requestNotificationsPermission = async () => {
-  switch (Platform.OS) {
-    case 'ios': {
-      const authStatus = await messaging().requestPermission();
-      return (authStatus === messaging.AuthorizationStatus.AUTHORIZED || authStatus === messaging.AuthorizationStatus.PROVISIONAL);
-    }
-    case 'android': {
-      const authStatus = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS);
-      return authStatus === 'granted';
-    }
-    default: {
-      return true;
-    }
+const ANDROID_NOTIFICATION_CHANNEL_ID = 'default';
+
+class NotificationHandler {
+  constructor() {
+    notifee.createChannel({
+      id: ANDROID_NOTIFICATION_CHANNEL_ID,
+      name: 'Default Channel',
+      importance: AndroidImportance.HIGH,
+    });
   }
-}
 
-export const onRemoteMessage = async remoteMessage => {
-  const currentScreen = navigationRef.current?.getCurrentRoute()?.name;
-  const isSendbirdNotification = Boolean(remoteMessage.data.sendbird);
-  if (!isSendbirdNotification) return;
+  startOnForeground() {
+    const unsubscribes = [];
 
-  const channelId = await notifee.createChannel({
-    id: 'SendbirdNotificationChannel',
-    name: 'Sendbird RN Notification Sample',
-  });
+    if (Platform.OS === 'android') {
+      messaging().onMessage(async remoteMessage => {
+        if (isSendbirdNotification(remoteMessage.data)) {
+          const sendbird = parseSendbirdNotification(remoteMessage.data);
+          const currentScreen = navigationRef.current?.getCurrentRoute()?.name;
 
-  if (remoteMessage && remoteMessage.data) {
-    let pushActionId = 'SendbirdNotification-';
-    const sendbirdData = JSON.parse(remoteMessage.data.sendbird);
+          if (
+            currentScreen === 'Notifications' &&
+            sendbird.channel.channel_url === store.getState().sendbird.feedChannel._url
+          ) {
+            dispatch(updateHasNewNotifications(true));
+          } else {
+            dispatch(refreshCollection());
+            await this.displayNotification(remoteMessage, sendbird);
+          }
+        }
+      });
+      unsubscribes.push(
+        notifee.onForegroundEvent(async ({type, detail}) => {
+          if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+            const payload = parseSendbirdNotification(detail.notification.data);
+          }
+        }),
+      );
```

Comment:
It seems not doing anything.

Reply:
If specifying the part that handles the notification when it's clicked, it would be a good idea to leave comments or logs!